flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractFlinkResourceReconciler.java

```diff
@@ -180,13 +190,28 @@ public void reconcile(FlinkResourceContext<CR> ctx) throws Exception {
         }
     }
 
-    private void applyAutoscaler(FlinkResourceContext<CR> ctx) throws Exception {
+    private void applyAutoscaler(FlinkResourceContext<CR> ctx, @Nullable String existingOverrides)
+            throws Exception {
         var autoScalerCtx = ctx.getJobAutoScalerContext();
         boolean autoscalerEnabled =
                 ctx.getResource().getSpec().getJob() != null
                         && ctx.getObserveConfig().getBoolean(AUTOSCALER_ENABLED);
         autoScalerCtx.getConfiguration().set(AUTOSCALER_ENABLED, autoscalerEnabled);
+
         autoscaler.scale(autoScalerCtx);
+
+        // Check that the overrides actually changed and not merely the String representation
+        var flinkConfig = ctx.getResource().getSpec().getFlinkConfiguration();
+        var newOverrides = flinkConfig.get(PipelineOptions.PARALLELISM_OVERRIDES.key());
```

Comment:
It’s not safe to access configs by key directly . We can use the ctx getDeployConfig and getObserveConfig to get the configs and then we can even avoid passing the extra parameter to the method

Reply:
I'm curious, why is it not safe?